### PR TITLE
Np 1979 migrate fetch publication

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: fetch-publication/build.gradle [hunk 1]
```diff
@@ -11,4 +11,21 @@ dependencies {
         exclude group: 'com.amazonaws', module: 'aws-java-sdk-core'
     }
     testImplementation project(':publication-testing')
+}
+
+
+configurations.testImplementation.canBeResolved = true
```
Comment (axthosarouris) [inline fetch-publication/build.gradle:17]: For running DynamoDB locally.

File: fetch-publication/src/test/java/no/unit/nva/publication/fetch/FetchPublicationHandlerTest.java [hunk 2]
```diff
@@ -143,19 +144,6 @@ public void handlerReturnsInternalServerErrorResponseOnUnexpectedException()
             MESSAGE_FOR_RUNTIME_EXCEPTIONS_HIDING_IMPLEMENTATION_DETAILS_TO_API_CLIENTS));
     }
 
-    @Test
-    @DisplayName("handler Returns BadGateway Response On Communication Problems")
-    public void handlerReturnsBadGatewayResponseOnCommunicationProblems()
```
Comment (axthosarouris) [inline fetch-publication/src/test/java/no/unit/nva/publication/fetch/FetchPublicationHandlerTest.java:148]: This exception is never thrown by DynamoDBService (old service) or the ResourceService (new service)

File: modify-publication/src/main/java/no/unit/nva/publication/modify/ModifyPublicationHandler.java [hunk 3]
```diff
@@ -66,12 +65,6 @@ protected PublicationResponse processInput(UpdatePublicationRequest input, Reque
         return PublicationMapper.convertValue(updatedPublication, PublicationResponse.class);
     }
 
-    private UserInstance extractUserInstance(RequestInfo requestInfo) {
```
Comment (axthosarouris) [inline modify-publication/src/main/java/no/unit/nva/publication/modify/ModifyPublicationHandler.java:69]: moved to RequestUtil

File: template.yaml [hunk 4]
```diff
@@ -149,6 +159,8 @@ Resources:
                 - application/json
               produces:
                 - application/json
+              security:
```
Comment (axthosarouris) [inline template.yaml:162]: GET publication was not secured before.

File: template.yaml [hunk 5]
```diff
@@ -158,7 +170,7 @@ Resources:
                   description: UUID identifier of the Publication to fetch.
               x-amazon-apigateway-integration:
                 uri:
-                  Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${NvaFetchPublicationFunction.Arn}/invocations
+                  Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${NvaFetchPublicationFunction.Arn}:live/invocations
```
Comment (axthosarouris) [inline template.yaml:173]: link to hot instance

File: template.yaml [hunk 6]
```diff
@@ -398,22 +408,49 @@ Resources:
...
         Variables:
           ALLOWED_ORIGIN: '*'
-          TABLE_NAME: !Ref PublicationsTableName
-          BY_PUBLISHER_INDEX_NAME: !Ref PublicationsByOwnerIndexName
-          BY_PUBLISHED_PUBLICATIONS_INDEX_NAME: !Ref PublishedPublicationsIndexName
+          TABLE_NAME: !Ref OrestisResources
       Policies:
         - DynamoDBCrudPolicy:
-            TableName: !Ref PublicationsTableName
+            TableName: !Ref OrestisResources
```
Comment (axthosarouris) [inline template.yaml:422]: Temporary table until the refactoring is complete and tested

File: template.yaml [hunk 7]
```diff
@@ -50,11 +50,21 @@ Parameters:
...
+    Type: 'AWS::SSM::Parameter::Value<String>'
     Default: nvaEventBusName
   EventBusArn:
-    Type : 'AWS::SSM::Parameter::Value<String>'
+    Type: 'AWS::SSM::Parameter::Value<String>'
     Default: nvaEventBusArn
+  MaxConcurrency:
+    Type: Number
+    Default: 100
+    Description: Max number of provisioned hot instnaces for a lambda function
```
Comment (brinxmat) [inline template.yaml:61]: Instnaces -> instances

**Review by brinxmat — APPROVED** [review]

**Pull request merged**